### PR TITLE
Hotfix/node 34 crypto updates

### PR DESCRIPTION
**Pull request by bizob2828**
This PR addresses the following:
 * Adds crypto rules to each framework with abstract sinks/views
 * Cleans up unused code in express
 * Makes some one off rules in hapi and express consistent with the abstract ones
 * PR for screener updates coming shortly

To test you should be able to go to any framework, run with agent and go to `/crypto`.  There are 3 different rules and you can make them unsafe and safe.  There isn't any user-controlled data, all the logic for determining if rule is unsafe/safe is in `test-bench-utils/lib/sinks/crypto.js`

### CODE REVIEW

File: express/views/partials/head.ejs [hunk 1]
```diff
@@ -1,6 +1,7 @@
 <meta charset="utf-8" />
 <meta http-equiv="X-UA-Compatible" content="IE=edge" />
 <meta name="viewport" content="width=device-width, initial-scale=1" />
+<meta http-equiv="Cache-Control" content="no-sleep">
```
Comment (tough-griff) [inline express/views/partials/head.ejs:4]: What's `no-sleep`? I don't see that option documented anywhere here:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control

File: test-bench-utils/lib/sinks/crypto.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+/**
+ * @param {string} not used but need to keep signature for abstracted helpers
+ * @param {Object} opts
+ * @param {boolean=} opts.safe are we calling the sink safely?
+ * @param {boolean=} opts.noop are we calling the sink as a noop?
+ */
+module.exports['crypto-bad-mac'] = async function exec(
+  input,
+  { safe = false, noop = false } = {}
+) {
```
Comment (tough-griff) [inline test-bench-utils/lib/sinks/crypto.js:13]: We could use the `_param` pattern here to indicate an unused argument.

```suggestion
/**
 * @param {string} _input not used but need to keep signature for abstracted helpers
 * @param {Object} opts
 * @param {boolean=} opts.safe are we calling the sink safely?
 * @param {boolean=} opts.noop are we calling the sink as a noop?
 */
module.exports['crypto-bad-mac'] = async function exec(
  _input,
  { safe = false, noop = false } = {}
) {
```

File: test-bench-utils/lib/sinks/crypto.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+  const algorithm = safe ? 'sha512' : 'md5';
+
+  return new Promise((resolve) => {
+    const hash = crypto.createHash(algorithm);
+    hash.update('salt');
+    const value = hash.digest('hex');
+    resolve(value);
+  });
+};
```
Comment (tough-griff) [inline test-bench-utils/lib/sinks/crypto.js:24]: This is in an `async function` so it's already being coerced to a `Promise`

```suggestion
  return crypto.createHash(algorithm)
    .update('salt')
    .digest('hex');
};
```

File: test-bench-utils/lib/sinks/crypto.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+ */
+module.exports['crypto-bad-ciphers'] = async function exec(
+  input,
+  { safe = false, noop = false } = {}
+) {
+  if (noop) return 'NOOP';
+
+  const { algorithm, bytes } = safe ? { algorithm: 'aes-256-cbc', bytes: 16 } : { algorithm: 'rc2', bytes: 8 };
+
+  return new Promise((resolve) => {
```
Comment (tough-griff) [inline test-bench-utils/lib/sinks/crypto.js:40]: see above.

File: test-bench-utils/lib/sinks/crypto.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+ * @param {boolean=} opts.noop are we calling the sink as a noop?
+ */
+module.exports['crypto-weak-randomness'] = async function exec(
+  input,
+  { safe = false, noop = false } = {}
+) {
+  if (noop) return 'NOOP';
+  const method = safe ? crypto.randomBytes(10).toString('hex') : Math.random(10).toString();
+
+  return new Promise((resolve) => {
```
Comment (tough-griff) [inline test-bench-utils/lib/sinks/crypto.js:64]: see above.

File: express/views/partials/head.ejs (hunk 1 above)
Comment (bizob2828) [reply express/views/partials/head.ejs:4]: I just added it to get `cache-controls-missing` firing as a POC.  I plan on following up with formal tickets to hit some of our missing rules

Comment (tough-griff) [reply express/views/partials/head.ejs:4]: Gotcha, so an intentionally invalid value.

**Pull request merged**